hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncScanSingleRegionRpcRetryingCaller.java

```diff
@@ -442,6 +449,32 @@ private void updateNextStartRowWhenError(Result result) {
     includeNextStartRowWhenError = result.mayHaveMoreCellsInRow();
   }
 
+  private boolean noMoreResultsForScan(Scan scan, RegionInfo info) {
+    if (isEmptyStopRow(info.getEndKey())) {
+      return true;
+    }
+    if (isEmptyStopRow(scan.getStopRow())) {
+      return false;
+    }
+    int c = comparator.compare(info.getEndKey(), scan.getStopRow());
```

Comment:
Why don't need comparator previously? Don't use this to scan meta previously?

Reply:
It is just because we only have one meta region in the past...